packages/logging/src/analytics-helpers.spec.ts

```diff
@@ -245,4 +249,51 @@ describe('analytics helpers', function () {
       ).to.match(/^(hi,hi,hi|bye,bye,bye)$/);
     });
   });
+
+  describe('SampledAnalytics', function () {
+    const userId = `u-${Date.now()}`;
+    const iEvt = { userId, traits: { platform: 'what', session_id: 'abc' } };
+    const tEvt = {
+      userId,
+      event: 'hi',
+      properties: { mongosh_version: '1.2.3', session_id: 'abc' },
+    };
+
+    afterEach(function () {
+      delete process.env.MONGOSH_ANALYTICS_SAMPLE;
+    });
+
+    it('should override sampling with the MONGOSH_ANALYTICS_SAMPLE environment variable', function () {
+      process.env.MONGOSH_ANALYTICS_SAMPLE = 'true';
```

Comment:
This should probably also be set for the existing tests for telemetry in the cli-repl package?

Reply:
It makes sense, testing it also at the cli-repl package.

Reply:
No, I mean, there are existing tests, right now they wouldn’t work because they would be subject to the sampling and therefore flaky, right?

Reply:
Ah yes, what I was thinking is on adding a beforeEach on all telemetry tests to enable the environment variable, and adding a test when it's disabled.